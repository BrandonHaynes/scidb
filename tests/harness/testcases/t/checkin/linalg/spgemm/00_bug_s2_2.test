--setup
load_library('linear_algebra')

--start-query-logging

create array sp1_c1_1  <a:double>[x=0:1,1,0, y=0:1,1,0]

--test
store( filter(build(sp1_c1_1,   1), true),  sp1_c1_1 )

spgemm(sp1_c1_1, sp1_c1_1,'rightReplicate=false')
spgemm(sp1_c1_1, sp1_c1_1,'rightReplicate=true')

--cleanup
remove(sp1_c1_1)

