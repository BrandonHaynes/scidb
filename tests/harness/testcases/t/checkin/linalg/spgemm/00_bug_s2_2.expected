SCIDB QUERY : <load_library('linear_algebra')>
Query was executed successfully

SCIDB QUERY : <create array sp1_c1_1  <a:double>[x=0:1,1,0, y=0:1,1,0]>
Query was executed successfully

SCIDB QUERY : <store( filter(build(sp1_c1_1,   1), true),  sp1_c1_1 )>
{x,y} a
{0,0} 1
{0,1} 1
{1,0} 1
{1,1} 1

SCIDB QUERY : <spgemm(sp1_c1_1, sp1_c1_1,'rightReplicate=false')>
{x,y} multiply
{0,0} 2
{0,1} 2
{1,0} 2
{1,1} 2

SCIDB QUERY : <spgemm(sp1_c1_1, sp1_c1_1,'rightReplicate=true')>
{x,y} multiply
{0,0} 2
{0,1} 2
{1,0} 2
{1,1} 2

SCIDB QUERY : <remove(sp1_c1_1)>
Query was executed successfully

